gammapy/cube/make.py

```diff
@@ -257,3 +260,125 @@ def _check_selection(selection):
             raise ValueError("Selection not available: {!r}".format(name))
 
     return selection
+
+class IrfMapMaker:
+    """Make IRF maps from IACT observations.
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.WcsGeom`
+        Reference map geometry with true energy axis
+    offset_max : `~astropy.coordinates.Angle`
+        Maximum offset angle
+    rad_axis : `~gammapy.maps.MapAxis`
+        The rad axis of the PsfMap to be built
+    migra_axis: `~gammapy.maps.MapAxis`
+        The migra axis of the EDispMap to be built
+    """
+
+    def __init__(self, geom, offset_max, rad_axis=None, migra_axis=None):
+        if not isinstance(geom, WcsGeom):
+            raise ValueError("IrfMapMaker only works with WcsGeom")
+
+        if geom.is_image:
+            raise ValueError("IrfMapMaker only works with geom with an energy axis")
+
+        self.geom = geom
+        self.offset_max = Angle(offset_max)
+
+        if rad_axis is None and migra_axis is None:
+            raise ValueError("No rad nor migra axis passed to IrfMapMaker")
+
+        self.rad_axis = rad_axis
+        self.migra_axis = migra_axis
+
+        self.maps = {}
+
+        # Initialise zero-filled maps
+        exposure = Map.from_geom(self.geom, unit="m2s")
+
+        energy_axis = self.geom.get_axis_by_name('energy')
+
+        if self.migra_axis is not None:
+            self.edisp_geom = self.geom.to_image().to_cube([migra_axis, energy_axis])
+            self.edisp_map = EDispMap(Map.from_geom(self.edisp_geom, unit=""),exposure)
+            self.maps['edisp'] = self.edisp_map
+
+        if self.rad_axis is not None:
+            self.psf_geom = self.geom.to_image().to_cube([rad_axis, energy_axis])
+            self.psf_map = PSFMap(Map.from_geom(self.psf_geom, unit="1/sr"), exposure)
+            self.maps['psf'] = self.psf_map
+
+
+    def run(self, observations):
+        """
+        Run IrfMapMaker for a list of observations to create
+        stacked PsfMap and EdispMap
+
+        Parameters
+        --------------
+        observations : `~gammapy.data.Observations`
+            Observations to process
+
+        Returns
+        -----------
+        maps: dict of stacked PsfMap and EdispMap
+        """
+        for obs in observations:
+            try:
+                self._process_obs(obs)
+            except NoOverlapError:
```

Comment:
I think currently no cutout is made. I the ideal implementation would be if `self._process_obs` computes `edisp`, `psf` and `exposure` map for one observation on a cutout. The stacking must the happen outside in the `.run()` method. I guess this means we can't re-use the `EDispMap.stack()` and `PSFMap.stack()` methods or these methods must accept an additional `geom` object, which defines a global geometry the cutouts a stacked into...

Reply:
What is needed is that `PSFMap.stack(other)` projects `other` on its own `geom`. This will allow to stack all individual `PSFMap` & `EDispMap` on the final ones. 

This is what is done in PR #2085.